### DISCUSSION TITLE
MariaDB

### DISCUSSION THREAD

**Discussion by mrmuskrat** [MaterializeInc/materialize#30699] [Feature Requests]
My boss asked me to check out Materialize for use in one of our applications. I started testing with the Docker image and was really liking it. It was only when I tried to test with one of dev databases that I realized that Materialize does not currently support MariaDB. Are there any plans to support MariaDB? If so, is there a rough timeline for when support will be added?

**Comment by morsapaes** [comment]
Thanks for opening a discussion around MariaDB support, @mrmuskrat! While designing the MySQL source, we made the conscious decision to not support MariaDB until we saw demand for it — MariaDB deviates enough from MySQL to require a non-zero degree of customization in our base implementation. For example:

* MariaDB uses a different GTID format than MySQL, which impacts how we handle snapshotting and resume operations.
* MariaDB supports data types that are not supported by MySQL, and handles some of the same data types differently (_e.g._ JSON).

All this being said: we will _eventually_ support it! Just not a short-term roadmap task, given the demand we've seen so far. But this thread is the right place to discuss and eventually bump its prioritization! In the meantime, you can still ingest MariaDB CDC into Materialize using [Kafka and Debezium](https://debezium.io/documentation/reference/stable/connectors/mariadb.html) (though it's...different), if that's something that could work for you!
